### ISSUE TITLE
Autocomplete - SelectedValueChanged fired automatically when typed in text equals the text in dropdown

### ISSUE THREAD

**Issue by ivancavlek** [Megabit/Blazorise#4245]
Let's say I have two items in `Data`, "item1" and "item2" (text is important, not values). I start typing and upon typing "item1" (exactly the text that is in `Data`), "item1" is highlighted in the dropdown and (here's the error) `SelectedValueChanged` is fired!
I haven't selected the dropdown value manually (with mouse click or on keyboard), it just fired automatically when the typed in text was exactly as the text in the dropdown. Any action afterwards in the dropdown (mouse or keyboard) on "item1" doesn't fire the `SelectedValueChanged`.

I believe this is an error, or am I missing something...

```C#
@inherits TextSearchBase

<Autocomplete TItem="ContainerItemDto"
              TValue="string"
              AutoPreSelect="false" SuggestSelectedItems="false" FreeTyping="true"
              AutoSelectFirstItem="false"
              Data="_repositoryContainerItems"
              Debounce="StoreApplicationUtility.DelayTextOnKeyPress"
              DebounceInterval="StoreApplicationUtility.DelayTextOnKeyPressInterval"
              Disabled="PickerIsDisabled"
              Filter="AutocompleteFilter.Contains"
              MinLength="StoreApplicationUtility.DelayTextMinLength"
              SearchChanged="OnSearchRepositoryContainerItems"
              SelectedValue="default"
              SelectedValueChanged="OnSelectRepositoryContainerItems"
              TextField="@(item => item?.DisplayName)"
              ValueField="@(item => item?.Id)">
    <NotFoundContent>
        @Localizer["Not found"]
    </NotFoundContent>
    <ItemContent>
        @if (PickerItemContent is null)
        {
            <Div>
                @context.Text
            </Div>
        }
        else
        {
            @PickerItemContent(context)
        }
    </ItemContent>
</Autocomplete>
```

**Comment by David-Moreira** [comment]
Hello,

I can confirm that the behaviour is now different.
What I have described is the one that's in **v1.0** and we'll be readjusting.

Basically `SelectedValueChanged` should ever only trigger 
- upon finding a value in the `Data` collection
- or going back to the `default` (null) value if it is not found

This is currently not happening like you've found out. Thanks for alterting us.

I also took the time to test `SelectedTextChanged`,
in **v1.0**, 
- it would always trigger upon text input with the currently selected text even if `FreeTyping` was not on.

in **v.1.1**, 
- If `FreeTyping` is on, it will always trigger with the current text input. 
- If `FreeTyping` is off it will always trigger with the `default` (null) upon text input, and with an actual `SelectedText `only upon finding a value in the `Data` collection.

Here on **v1.1** this is the correct behaviour, however I don't see the need for `FreeTyping="false"` to keep triggering `SelectedTextChanged` with a `default` (null) value every single time. The input has changed but the `SelectedText` hasn't changed at all.

So in summary, what we'll be readjusting in this issue's scope:
- `SelectedValueChanged` only triggers when appropriate (find an actual value, or going back to `default`)
- `SelectedTextChanged` + `FreeTyping=="false"` only triggers upon an actual `SelectedText` or going back to the `default` (null), just like the `SelectedValue`.

**Comment by stsrki** [comment]
@David-Moreira sounds good to me.
